File: mobile/hika_app/lib/features/drivers/data/vehicle.dart
class VehiclePhoto {
  const VehiclePhoto({required this.id, required this.url, required this.isPrimary});

  factory VehiclePhoto.fromJson(Map<String, dynamic> json) =>
      VehiclePhoto(id: json['id'] as String, url: json['url'] as String, isPrimary: json['isPrimary'] as bool);

  final String id;
  final String url;
  final bool isPrimary;
}

class Vehicle {
  const Vehicle({
    required this.id,
    required this.make,
    required this.model,
    required this.year,
    required this.color,
    required this.registrationNumber,
    required this.seatCapacity,
    required this.isVerified,
    required this.photos,
  });

  factory Vehicle.fromJson(Map<String, dynamic> json) => Vehicle(
    id: json['id'] as String,
    make: json['make'] as String,
    model: json['model'] as String,
    year: json['year'] as int,
    color: json['color'] as String,
    registrationNumber: json['registrationNumber'] as String,
    seatCapacity: json['seatCapacity'] as int,
    isVerified: json['isVerified'] as bool,
    photos: (json['photos'] as List<dynamic>? ?? [])
        .map((p) => VehiclePhoto.fromJson(p as Map<String, dynamic>))
        .toList(),
  );

  final String id;
  final String make;
  final String model;
  final int year;
  final String color;
  final String registrationNumber;
  final int seatCapacity;
  final bool isVerified;
  final List<VehiclePhoto> photos;

  String get displayName => '$year $make $model';

  VehiclePhoto? get primaryPhoto {
    for (final photo in photos) {
      if (photo.isPrimary) {
        return photo;
      }
    }
    return photos.isEmpty ? null : photos.first;
  }
}

File: backend/src/Hika.Domain/Drivers/Vehicle.cs
using Hika.Domain.Common;

namespace Hika.Domain.Drivers;

public sealed class Vehicle : AuditableEntity
{
    public Guid DriverProfileId { get; private set; }

    public string Make { get; private set; }

    public string Model { get; private set; }

    public int Year { get; private set; }

    public string Color { get; private set; }

    public string RegistrationNumber { get; private set; }

    public int SeatCapacity { get; private set; }

    /// <summary>Cached from the latest Verification of type VehicleRegistration.</summary>
    public bool IsVerified { get; private set; }

    private readonly List<VehiclePhoto> _photos = [];
    public IReadOnlyCollection<VehiclePhoto> Photos => _photos.AsReadOnly();

    private Vehicle()
    {
        Make = string.Empty;
        Model = string.Empty;
        Color = string.Empty;
        RegistrationNumber = string.Empty;
    }

    private Vehicle(
        Guid driverProfileId, string make, string model, int year, string color,
        string registrationNumber, int seatCapacity)
    {
        DriverProfileId = driverProfileId;
        Make = make;
        Model = model;
        Year = year;
        Color = color;
        RegistrationNumber = registrationNumber;
        SeatCapacity = seatCapacity;
        IsVerified = false;
    }

    public static Vehicle Create(
        Guid driverProfileId, string make, string model, int year, string color,
        string registrationNumber, int seatCapacity) =>
        new(driverProfileId, make, model, year, color, registrationNumber, seatCapacity);

    public void Update(string make, string model, int year, string color, string registrationNumber, int seatCapacity)
    {
        if (registrationNumber != RegistrationNumber)
        {
            // A changed registration number is effectively a different vehicle for
            // verification purposes — the prior verification no longer applies.
            IsVerified = false;
        }

        Make = make;
        Model = model;
        Year = year;
        Color = color;
        RegistrationNumber = registrationNumber;
        SeatCapacity = seatCapacity;
    }

    public void MarkVerified() => IsVerified = true;

    public void MarkUnverified() => IsVerified = false;

    public VehiclePhoto AddPhoto(string url, bool isPrimary)
    {
        if (isPrimary)
        {
            foreach (var existing in _photos)
            {
                existing.SetPrimary(false);
            }
        }

        var photo = new VehiclePhoto(Id, url, isPrimary, _photos.Count);
        _photos.Add(photo);
        return photo;
    }
}
